### PR TITLE
Use new svyquantile

### PR DESCRIPTION
**Pull request by bschneidr**
This update makes sure that the new `svyquantile()` works correctly and has the changed parameters documented. The bulk of the updates are simply to make sure the tests are working, since the output format for the survey package has changed.

I still need to add in the `oldsvyquantile()` function that can be used for backwards compatibility.

Addresses #101. Also related to #119, the new default degrees of freedom for confidence intervals with `svyquantile()` is `degf(svy)` rather than `Inf`, so I made note of that in the function's documentation.

### CODE REVIEW

File: R/survey_statistics.r [hunk 1]
```diff
@@ -363,11 +360,10 @@ survey_quantile <- function(
   if (!is.null(vartype)) {
     vartype <- if (missing(vartype)) "se" else match.arg(vartype, several.ok = TRUE)
   }
-  if (missing(interval_type) & !inherits(.svy, "svyrep.design")) interval_type <- "Wald"
-  if (missing(interval_type) & inherits(.svy, "svyrep.design")) interval_type <- "probability"
+  if (missing(interval_type) & !inherits(.svy, "svyrep.design")) interval_type <- "mean"
```
Comment (gergness) [inline R/survey_statistics.r:362]:
```suggestion
  if (missing(interval_type) & !inherits(.svy, "svyrep.design")) interval_type <- "mean"
```

Comment (gergness) [inline R/survey_statistics.r:362]: Now that all types of surveys use "mean" as default, collapse to a single line: 
```
if (missing(interval_type)) interval_type <- "mean"
```

File: R/survey_statistics.r [hunk 2]
```diff
@@ -363,11 +360,10 @@ survey_quantile <- function(
...
+  if (missing(interval_type) & inherits(.svy, "svyrep.design")) interval_type <- "mean"
```
Comment (gergness) [inline R/survey_statistics.r:363]: Now that all types of surveys use "mean" as default, collapse to a single line: 
```
if (missing(interval_type)) interval_type <- "mean"
```

File: R/survey_statistics.r [hunk 3]
```diff
@@ -420,20 +409,20 @@ survey_median <- function(
   if (!is.null(vartype)) {
     vartype <- if (missing(vartype)) "se" else match.arg(vartype, several.ok = TRUE)
   }
-  if (missing(interval_type) & !inherits(.svy, "svyrep.design")) interval_type <- "Wald"
-  if (missing(interval_type) & inherits(.svy, "svyrep.design")) interval_type <- "probability"
+  if (missing(interval_type) & !inherits(.svy, "svyrep.design")) interval_type <- "mean"
```
Comment (gergness) [inline R/survey_statistics.r:411]: Single line again here

```
if (missing(interval_type)) interval_type <- "mean"
```

File: tests/testthat/test_survey_statistics.r [hunk 4]
```diff
@@ -1,11 +1,6 @@
 context("Quick tests for summary stats (ratio / quantile)")
 
-# TODO: switch to improved svyquantile
-if (utils::packageVersion("survey") >= "4.1") {
-  svyq_func <- get("oldsvyquantile", asNamespace("survey"))
-} else {
-  svyq_func <- survey::svyquantile
-}
+svyq_func <- survey::svyquantile
```
Comment (gergness) [inline tests/testthat/test_survey_statistics.r:3]: Can you change the places that used `svyq_func` to just be svyquantile now (and then delete this line)? (Like you did in the other test file)

**Review by gergness — CHANGES_REQUESTED** [review]
Thanks so much! It looks like the remaining github action failures are going to require debugging github actions (it's weird how the logs are empty). I'm happy to take care of that, unless you want to go down that rabbithole. 

Do you think you'll have time for `old_survey_quantile`? Or do you want me to do that?

File: R/survey_statistics.r (hunk 2 above)
Comment (bschneidr) [reply R/survey_statistics.r]: Glad you noticed that. https://github.com/gergness/srvyr/pull/124/commits/f1a4cd92f8c8c03e9766495b5e0299ef6fc77d7b ought to fix this.

File: R/survey_statistics.r (hunk 3 above)
Comment (bschneidr) [reply R/survey_statistics.r]: Should also be resolved by https://github.com/gergness/srvyr/pull/124/commits/f1a4cd92f8c8c03e9766495b5e0299ef6fc77d7b.

File: tests/testthat/test_survey_statistics.r (hunk 4 above)
Comment (bschneidr) [reply tests/testthat/test_survey_statistics.r]: Another one resolved by https://github.com/gergness/srvyr/pull/124/commits/f1a4cd92f8c8c03e9766495b5e0299ef6fc77d7b.

**Comment by bschneidr** [comment]
> Thanks so much! It looks like the remaining github action failures are going to require debugging github actions (it's weird how the logs are empty). I'm happy to take care of that, unless you want to go down that rabbithole.
> 
> Do you think you'll have time for `old_survey_quantile`? Or do you want me to do that?

I think I can tackle the programming and roxygen2 documentation for `old_survey_quantile` (I'm thinking I'll just copy the old code/documentation and insert it below `survey_median()`, then update the documentation to note the changes flowing from the `survey` package and link to descriptions of the update). But I think it would be great if you were to start writing something for users about the change, maybe in `NEWS`?

**Comment by gergness** [comment]
Sounds good! Thanks again!

**Comment by gergness** [comment]
Ha, yeah it's an unusual need, I think what you did makes sense though.

**Pull request merged**